### DISCUSSION TITLE
background gradient duration

### DISCUSSION THREAD

**Discussion by CFarhad** [tailwindlabs/tailwindcss#6614] [Help] [answered]
hello.
i have important question. there is a button and it has default gradient background with `bg-gradient-to-r from-cyan-500 to-blue-500`;
i want if button is hover gradient background change to `hover:from-blue-500 hover:to-cyan-500` slow , but not working.

who can help me guys

**Comment by NovusOpus** [comment, marked as answer]
This is not posible without a work around. Since currently gradients don't support transitions. [This is not just for Tailwind CSS but also for CSS in general](https://developer.mozilla.org/en-US/docs/Web/CSS/transition#browser_compatibility).

One way to archive this, is to double the background size of the element and transition the background position. 

By extending `backgroundSize`  and `backgroundPosition` in our `tailwind.config`. [Like this](https://play.tailwindcss.com/cR7wyHyoLK?file=config):
```JS
module.exports = {
  theme: {
    extend: {
      backgroundSize: {
        'size-200': '200% 200%',
      },
      backgroundPosition: {
        'pos-0': '0% 0%',
        'pos-100': '100% 100%',
      },
    },
  },
  plugins: [],
}
```
And the button should be like this:

```html
<button class="p-10 text-white rounded duration-500 bg-gradient-to-r from-cyan-500 via-blue-500 to-cyan-500 bg-size-200 bg-pos-0 hover:bg-pos-100">Hover me</button>
```

#### Note that we use the [`via-{color} Gradient Color Stops`](https://tailwindcss.com/docs/gradient-color-stops#middle-color), to get the desired gradient side swap.

## Working Example: [Tailwind Play](https://play.tailwindcss.com/cR7wyHyoLK)

Sources: [Stack Overflow -Tailwind Gradient Transition](https://stackoverflow.com/questions/67150736/tailwind-background-gradient-transition), [Stack Overflow - CSS Gradient Transitions](https://stackoverflow.com/questions/6542212/use-css3-transitions-with-gradient-backgrounds)

Update: Added an alternative Way Shared by Tailwind CSS Staff.
---
Another way is to have an "overlay" gradient with different colors, and animate the opacity of that overlay on hover.
Like @adamwathan explains in this [commend](https://github.com/tailwindlabs/tailwindcss/discussions/6154#discussioncomment-1677029), or Like @simonswiss example in [another commend](https://github.com/tailwindlabs/tailwindcss/discussions/4145#discussioncomment-647917)

**Reply by newbelaravel** [reply]
In case anyone wants in tailwind v4
https://play.tailwindcss.com/HltHV6nx4Z
Much appreciated!
